### PR TITLE
Implement marshalling and unmarshalling of asset IDs.

### PR DESCRIPTION
**Pull request by derbear**
## Summary

This PR implements marshaling and unmarshaling of asset IDs as `<addr>/<asset ID>`.

## Test Plan

Tests live in `data/basics/asset_test.go`.

### CODE REVIEW

File: data/basics/userBalance.go [hunk 1]
```diff
@@ -281,3 +283,38 @@ type BalanceRecord struct {
...
+}
+
+// UnmarshalText initializes the assetID from an array of bytes.
+func (id *AssetID) UnmarshalText(text []byte) error {
+	address, err := UnmarshalChecksumAddress(string(text[:58]))
+	if err != nil {
+		return err
+	}
+
+	index, err := strconv.ParseUint(string(text[59:]), 10, 0)
```
[the remark this answers is not in the document]
Comment (tsachiherman) [reply data/basics/userBalance.go]: btw -  would it crash if the input string contains just the address ?

[the remark this answers is not in the document]
Comment (zeldovich) [reply data/basics/userBalance.go]: Regarding the vanity asset comment, the user has relatively little control over the index part, so I don't think they'd be able to make the string be too interesting-looking when formatted as base36.

File: data/basics/userBalance.go [hunk 2]
```diff
@@ -281,3 +283,38 @@ type BalanceRecord struct {
...
+}
+
+// MarshalText returns the full assetID as an array of bytes.
+func (id AssetID) MarshalText() ([]byte, error) {
+	return []byte(id.String()), nil
+}
+
+// UnmarshalText initializes the assetID from an array of bytes.
+func (id *AssetID) UnmarshalText(text []byte) error {
+	address, err := UnmarshalChecksumAddress(string(text[:58]))
```
[the remark this answers is not in the document]
Comment (rotemh) [reply data/basics/userBalance.go]: Also, very nit but consider making the magic numbers constants.

**Comment by rotemh** [comment]
General comment - I believe that we should not merge this PR. Creating another convention for our address with every new feature may create inconsistencies. A longterm fundamental solution would be to implement Zhenfei's or Zhenfei's like address scheme.

For now, I believe that making sending assets easier for the user can be done at the UI level and at Algod.

**Comment by derbear** [comment]
Sounds good; I also think we'll be replacing the 40-byte asset IDs with 8-byte asset IDs. I'll reopen later if we want this for some reason.

**Pull request closed without merging**